### PR TITLE
Balance patch

### PR DESCRIPTION
**Pull request by OtakuFreak13**
Changed the pricing and balance of some of the upgrades.

### CODE REVIEW

**Review by fredrix96 — DISMISSED** [review]
Seems more balanced

File: StortSpelProjekt/Game/src/Gamefiles/Components/UpgradeComponents/Upgrades/UpgradeMeleeDamage.cpp [hunk 1]
```diff
@@ -32,7 +32,7 @@ void UpgradeMeleeDamage::IncreaseLevel()
 	// increase level
 	m_Level++;
 
-	m_Price = m_StartingPrice + m_StartingPrice * ((float)m_Level / 2.0f);
+	m_Price = m_StartingPrice * (m_Level + 1);
 	// now apply the new stat
```
Comment (Schtekt) [inline StortSpelProjekt/Game/src/Gamefiles/Components/UpgradeComponents/Upgrades/UpgradeMeleeDamage.cpp]: Please remove this comment, it does not tell the reader anything.

File: StortSpelProjekt/Game/src/Gamefiles/Components/UpgradeComponents/Upgrades/UpgradeMeleeLifeSteal.cpp [hunk 2]
```diff
@@ -1,14 +1,13 @@
...
 #include "Components/MeleeComponent.h"
 
-// Gives 10% lifesteal to melee attacks. Subsequent levels will add 5% more lifesteal. 
-// This procentage is based on the melee damage you do so 50 meleeDamage will give 5hp in lifesteal.
-// Price scaling should be 300 -> 700 -> 1300 -> 2100 -> 3100 -> 4300 -> 5700 -> 7300 -> 9100 etc.
-// Lifesteal at these lvls 10% -> 15% -> 20%  -> 25%  -> 30%  -> 35%  -> 40%  -> 45%  -> 50%  etc.
+// Gives 20% lifesteal to melee attacks. Subsequent levels will add 5% more lifesteal. 
+// This procentage is based on the melee damage you do so 50 meleeDamage will give 10hp in lifesteal.
+// Price scaling should be 300 -> 500 -> 700 -> 900 -> 1100 -> 1300 -> 1500 -> 1700 -> 1900 -> 2100 -> 2300 -> 2500 -> 2700 -> 2900 -> 3100 -> 3300 -> 3500 etc.
+// Lifesteal at these lvls 20% -> 25% -> 30% -> 35% -> 40%  -> 45%  -> 50%  -> 55%  -> 60%  -> 65%  -> 70%  -> 75%  -> 80%  -> 85%  -> 90%  -> 95%  -> 100%  etc.
```
Comment (Schtekt) [inline StortSpelProjekt/Game/src/Gamefiles/Components/UpgradeComponents/Upgrades/UpgradeMeleeLifeSteal.cpp]: The calculations and comments below show that the lifesteal percentage starts at 10%. Please decide on which you want to use.

File: StortSpelProjekt/Game/src/Gamefiles/Components/UpgradeComponents/Upgrades/UpgradePoisonAttack.cpp [hunk 3]
```diff
@@ -33,8 +34,11 @@ void UpgradePoisonAttack::IncreaseLevel()
 {
 	m_Level++;
 	m_NrOfTicks = 9 + m_Level;
-	m_Damage = (float)(0.04 * m_Level);
-	m_Slow = 0.10 + (float)m_Level / 10.0f;
+	// -1 because the first level should start at 0.04
```
Comment (Schtekt) [inline StortSpelProjekt/Game/src/Gamefiles/Components/UpgradeComponents/Upgrades/UpgradePoisonAttack.cpp]: Please remove this comment. One should be able to see this in the calculations. It is also not mentioned in other upgrades. If it must remain, please add the same comment to the rest of the upgrades.

File: StortSpelProjekt/Game/src/Gamefiles/Components/UpgradeComponents/Upgrades/UpgradePoisonAttack.cpp [hunk 4]
```diff
@@ -33,8 +34,11 @@ void UpgradePoisonAttack::IncreaseLevel()
...
+	m_Damage = 0.04 + (float)(0.005 * (m_Level - 1));
+	// Slow of 50% at maxlvlso enemies aren't snails.
+	// -1 because the first level should start at 0.10.
```
Comment (Schtekt) [inline StortSpelProjekt/Game/src/Gamefiles/Components/UpgradeComponents/Upgrades/UpgradePoisonAttack.cpp]: Same as above.

**Review by Schtekt — CHANGES_REQUESTED** [review]
Goodie!
I have left some comments about comments that I want to see changed, but otherwise good.

**Review by Schtekt — APPROVED** [review]

**Review by fredrix96 — APPROVED** [review]

**Pull request merged**